server/sonar-web/src/main/js/api/projectActivity.ts

```diff
@@ -17,53 +17,42 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-// @flow
-import { getJSON, postJSON, post } from '../helpers/request';
+import { getJSON, postJSON, post, RequestData } from '../helpers/request';
 import throwGlobalError from '../app/utils/throwGlobalError';
 
-/*::
-type GetProjectActivityResponse = {
-  analyses: Array<Object>,
+interface IGetProjectActivityResponse {
+  analyses: any[];
   paging: {
-    total: number,
-    pageIndex: number,
-    pageSize: number
-  }
-};
-*/
-
-/*::
-type GetProjectActivityOptions = {
-  project: string,
-  category?: ?string,
-  p?: ?number,
-  ps?: ?number
-};
-*/
+    total: number;
+    pageIndex: number;
+    pageSize: number;
+  };
+}
 
-export function getProjectActivity(
-  data /*: GetProjectActivityOptions */
-) /*: Promise<GetProjectActivityResponse> */ {
+export function getProjectActivity(data: {
+  project: string;
+  category?: string;
+  p?: number;
+  ps?: number;
+}): Promise<IGetProjectActivityResponse> {
```

Comment:
So far TS is unable to track types for fulfilled and rejected promises separately. So there is no much sense to add `| Response` here, it won't give better types. I've added it only where TS was complaining.